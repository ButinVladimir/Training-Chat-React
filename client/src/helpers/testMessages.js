import Message from './Message';

const testMessages = [
  new Message('1', '8/28/2018', 'User 1', [], 'Message 1'),
  new Message('2', '8/28/2018', 'User 1', [], 'Message 2'),
  new Message('3', '8/28/2018', 'User 1', ['User 2'], 'Message 3 to one user'),
  new Message('4', '8/28/2018', 'User 2', ['User 1', 'User 2'], 'User 2 answer'),
  new Message('5', '8/28/2018', 'User 3', ['User 1', 'User 2'], 'User 3 question?'),
  new Message('6', '8/28/2018', 'User 1', ['User 1', 'User 2', 'User 3'], 'User 1 answer'),
  new Message('7', '8/28/2018', 'User 2', [], 'Message 7'),
  new Message('8', '8/28/2018', 'User 2', [], 'Message 8'),
  new Message('9', '8/28/2018', 'User 1', [], 'Message 9'),
  new Message('10', '8/28/2018', 'User 3', [], 'Message 10'),
  new Message('11', '8/28/2018', 'User 1', [], 'Message 1'),
  new Message('12', '8/28/2018', 'User 1', [], 'Message 2'),
  new Message('13', '8/28/2018', 'User 1', ['User 2'], 'Message 3 to one user'),
  new Message('14', '8/28/2018', 'User 2', ['User 1', 'User 2'], 'User 2 answer'),
  new Message('15', '8/28/2018', 'User 3', ['User 1', 'User 2'], 'User 3 question?'),
  new Message('16', '8/28/2018', 'User 1', ['User 1', 'User 2', 'User 3'], 'User 1 answer'),
  new Message('17', '8/28/2018', 'User 2', [], 'Message 7'),
  new Message('18', '8/28/2018', 'User 2', [], 'Message 8'),
  new Message('19', '8/28/2018', 'User 1', [], 'Message 9'),
  new Message('20', '8/28/2018', 'User 3', [], 'Message 10'),
  new Message('21', '8/28/2018', 'User 1', [], 'Message 1'),
  new Message('22', '8/28/2018', 'User 1', [], 'Message 2'),
  new Message('23', '8/28/2018', 'User 1', ['User 2'], 'Message 3 to one user'),
  new Message('24', '8/28/2018', 'User 2', ['User 1', 'User 2'], 'User 2 answer'),
  new Message('25', '8/28/2018', 'User 3', ['User 1', 'User 2'], 'User 3 question?'),
  new Message('26', '8/28/2018', 'User 1', ['User 1', 'User 2', 'User 3'], 'User 1 answer'),
  new Message('27', '8/28/2018', 'User 2', [], 'Message 7'),
  new Message('28', '8/28/2018', 'User 2', [], 'Message 8'),
  new Message('29', '8/28/2018', 'User 1', [], 'Message 9'),
  new Message('30', '8/28/2018', 'User 3', [], 'Message 10'),
];

export default testMessages;
