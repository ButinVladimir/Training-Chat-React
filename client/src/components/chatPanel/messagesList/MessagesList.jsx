import React from 'react';
import PropTypes from 'prop-types';
import Message from './MessageContainer';
import { messageShape } from '../../../helpers/Message';
import './style.css';

function MessagesList({ messages }) {
  const convertedMessages = messages.map(message => (
    <Message key={message.id} {...message} />
  ));

  return (
    <div className="chat-messages-list-container">
      <ul className="chat-messages-list">
        {convertedMessages}
      </ul>
    </div>
  );
}

MessagesList.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape(messageShape)).isRequired,
};

export default MessagesList;
