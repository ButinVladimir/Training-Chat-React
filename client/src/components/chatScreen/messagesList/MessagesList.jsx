import React from 'react';
import PropTypes from 'prop-types';
import Message from './Message';
import { messageShape } from '../../../helpers/Message';
import './style.css';

function MessagesList({ messages, onClickUser }) {
  const convertedMessages = messages.map(message => (
    <Message key={message.id} {...message} onClickUser={onClickUser} />
  ));

  return (
    <div className="chat-messages-list-container">
      <ul className="chat-messages-list">
        {convertedMessages}
      </ul>
    </div>
  );
}

MessagesList.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape(messageShape)).isRequired,
  onClickUser: PropTypes.func.isRequired,
};

export default MessagesList;
