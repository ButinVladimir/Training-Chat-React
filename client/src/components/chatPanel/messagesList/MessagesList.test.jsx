import React from 'react';
import { shallow } from 'enzyme';
import MessagesList from './MessagesList';
import testMessages from '../../../helpers/testMessages';

describe('MessagesList', () => {
  it('renders without crashing', () => {
    const component = shallow(<MessagesList messages={testMessages} />);
    expect(component).toMatchSnapshot();
  });
});
