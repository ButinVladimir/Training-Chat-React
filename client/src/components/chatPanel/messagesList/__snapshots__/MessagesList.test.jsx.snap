// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesList renders without crashing 1`] = `
<div
  className="chat-messages-list-container"
>
  <ul
    className="chat-messages-list"
  >
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="1"
      key="1"
      message="Message 1"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="2"
      key="2"
      message="Message 2"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="3"
      key="3"
      message="Message 3 to one user"
      to={
        Array [
          "User 2",
        ]
      }
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 2"
      id="4"
      key="4"
      message="User 2 answer"
      to={
        Array [
          "User 1",
          "User 2",
        ]
      }
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 3"
      id="5"
      key="5"
      message="User 3 question?"
      to={
        Array [
          "User 1",
          "User 2",
        ]
      }
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="6"
      key="6"
      message="User 1 answer"
      to={
        Array [
          "User 1",
          "User 2",
          "User 3",
        ]
      }
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 2"
      id="7"
      key="7"
      message="Message 7"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 2"
      id="8"
      key="8"
      message="Message 8"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="9"
      key="9"
      message="Message 9"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 3"
      id="10"
      key="10"
      message="Message 10"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="11"
      key="11"
      message="Message 1"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="12"
      key="12"
      message="Message 2"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="13"
      key="13"
      message="Message 3 to one user"
      to={
        Array [
          "User 2",
        ]
      }
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 2"
      id="14"
      key="14"
      message="User 2 answer"
      to={
        Array [
          "User 1",
          "User 2",
        ]
      }
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 3"
      id="15"
      key="15"
      message="User 3 question?"
      to={
        Array [
          "User 1",
          "User 2",
        ]
      }
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="16"
      key="16"
      message="User 1 answer"
      to={
        Array [
          "User 1",
          "User 2",
          "User 3",
        ]
      }
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 2"
      id="17"
      key="17"
      message="Message 7"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 2"
      id="18"
      key="18"
      message="Message 8"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="19"
      key="19"
      message="Message 9"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 3"
      id="20"
      key="20"
      message="Message 10"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="21"
      key="21"
      message="Message 1"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="22"
      key="22"
      message="Message 2"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="23"
      key="23"
      message="Message 3 to one user"
      to={
        Array [
          "User 2",
        ]
      }
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 2"
      id="24"
      key="24"
      message="User 2 answer"
      to={
        Array [
          "User 1",
          "User 2",
        ]
      }
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 3"
      id="25"
      key="25"
      message="User 3 question?"
      to={
        Array [
          "User 1",
          "User 2",
        ]
      }
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="26"
      key="26"
      message="User 1 answer"
      to={
        Array [
          "User 1",
          "User 2",
          "User 3",
        ]
      }
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 2"
      id="27"
      key="27"
      message="Message 7"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 2"
      id="28"
      key="28"
      message="Message 8"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 1"
      id="29"
      key="29"
      message="Message 9"
      to={Array []}
    />
    <Connect(Message)
      date="8/28/2018"
      from="User 3"
      id="30"
      key="30"
      message="Message 10"
      to={Array []}
    />
  </ul>
</div>
`;
